src/dataflow-types/src/types.rs

```diff
@@ -1418,6 +1424,12 @@ pub mod sources {
         pub channel: String,
     }
 
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub struct LokiSourceConnector {
+        pub address: String,
+        pub query: String,
+    }
```

Comment:
Are we going to provide authentication through environment variables? S3 [uses](https://materialize.com/docs/sql/create-source/s3/#aws-credentials) the AWS credentials chain. We could use the same in the future (see https://github.com/grafana/loki/issues/5453)

Reply:
Looks like there's a recent design doc for secrets: https://github.com/MaterializeInc/materialize/pull/11022 so maybe best not to overthink this one for now.